### PR TITLE
Allow user to modify gPTP BMCA priority1 and priority2 variables

### LINKED ISSUE #28151: gPTP should allow user setting of priority1 and priority2 fields used in BMCA
**Describe the bug**
The BMCA (Best Master Clock selection Algorithm) uses priority1 and priority2 parameters. Currently the value of these is hardcoded and cannot be changed by the user. It should be possible to user to tweak them as needed. One way to allow tweaking is to provide the values as Kconfig options.

### PR DESCRIPTION
**Pull request by jukkar**
Fixes #28151

### CODE REVIEW

File: subsys/net/l2/ethernet/gptp/Kconfig [hunk 1]
```diff
@@ -195,6 +195,31 @@ config NET_GPTP_PATH_TRACE_ELEMENTS
 	  Announce message. Each array element takes 8 bytes. If this value
 	  is set to 8, then 8 * 8 = 64 bytes of memory is used.
 
+config NET_GPTP_BMCA_PRIORITY1
+	int "BMCA priority1 value"
+	default 248 if NET_GPTP_GM_CAPABLE
+	default 255
+	range 0 255
+	help
+	  The  priority1 attribute of the local clock. It is used in the
```
Comment (rlubos) [inline subsys/net/l2/ethernet/gptp/Kconfig:204]: nit: Double space after `The`

File: subsys/net/l2/ethernet/gptp/Kconfig [hunk 2]
```diff
@@ -195,6 +195,31 @@ config NET_GPTP_PATH_TRACE_ELEMENTS
...
+	  See Chapter 8.6.2.1 of IEEE 802.1AS for a more detailed description
+	  of priority1. Note that if the system	is non GM capable, then the
+	  value 255 is used always and this setting is ignored.
+
+config NET_GPTP_BMCA_PRIORITY2
+	int "BMCA priority2 value"
+	default 248
+	range 0 255
+	help
+	  The  priority2 attribute of the local clock. It is used in the BMCA
```
Comment (rlubos) [inline subsys/net/l2/ethernet/gptp/Kconfig:217]: nit: Double space after `The`

**Review by rlubos — APPROVED** [review]
LGTM

File: subsys/net/l2/ethernet/gptp/Kconfig (hunk 2 above)
Comment (jukkar) [reply subsys/net/l2/ethernet/gptp/Kconfig:217]: Doh! Will fix :)

File: subsys/net/l2/ethernet/gptp/Kconfig [hunk 3]
```diff
@@ -195,6 +195,31 @@ config NET_GPTP_PATH_TRACE_ELEMENTS
...
+	default 255
+	range 0 255
+	help
+	  The priority1 attribute of the local clock. It is used in the
+	  Best Master Clock selection Algorithm (BMCA), lower values take
+	  precedence. The default value is 255 if the device is non grand
+	  master capable, and 248 if it is GM capable.
+	  See Chapter 8.6.2.1 of IEEE 802.1AS for a more detailed description
+	  of priority1. Note that if the system	is non GM capable, then the
+	  value 255 is used always and this setting is ignored.
```
Comment (mbolivar-nordic) [inline subsys/net/l2/ethernet/gptp/Kconfig:210]: Is that really correct? It seems like it defaults to 255 if CONFIG_NET_GPTP_GM_CAPABLE=n, but this setting is not ignored. Am I missing something?

If it is important to really ignore this, why not make this option `depends on NET_GPTP_GM_CAPABLE` and simply have `default 248`? It seems gptp.h could remain as-is in this PR if that is done.

Comment (jukkar) [reply subsys/net/l2/ethernet/gptp/Kconfig:210]: The priority1 field does not depend on GM capable flag and user is free to set the priority field as he/she wishes (except value 255).

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**